src/client/eppo-client.ts

```diff
@@ -331,20 +355,55 @@ export default class EppoClient implements IEppoClient {
 
     // Compute variation for subject.
     const { subjectShards } = experimentConfig;
-    const { variations } = allocation;
-
-    const shard = getShard(`assignment-${subjectKey}-${flagKey}`, subjectShards);
-    const assignedVariation = variations.find((variation) =>
-      isShardInRange(shard, variation.shardRange),
-    );
+    const { variations, holdouts, statusQuoVariationKey, shippedVariationKey } = allocation;
+
+    let assignedVariation: IVariation | undefined;
+    let holdoutVariation = null;
+
+    const holdoutShard = getShard(`holdout-${subjectKey}`, subjectShards);
+    const matchingHoldout = holdouts.find((holdout) => {
+      const { statusQuoShardRange, shippedShardRange } = holdout;
+      if (isShardInRange(holdoutShard, statusQuoShardRange)) {
+        assignedVariation = variations.find(
+          (variation) => variation.variationKey === statusQuoVariationKey,
+        );
+        // Only log the holdout variation if this is a rollout allocation
+        // Only rollout allocations have shippedShardRange specified
+        if (shippedShardRange) {
+          holdoutVariation = HoldoutVariationEnum.STATUS_QUO;
+        }
+      } else if (shippedShardRange && isShardInRange(holdoutShard, shippedShardRange)) {
+        assignedVariation = variations.find(
+          (variation) => variation.variationKey === shippedVariationKey,
+        );
+        holdoutVariation = HoldoutVariationEnum.ALL_SHIPPED;
+      }
+      return assignedVariation;
+    });
+    let holdoutKey = null;
+    if (matchingHoldout) {
+      holdoutKey = matchingHoldout.holdoutKey;
```

Comment:
```suggestion
    const holdoutKey = matchingHoldout?.holdoutKey;
```

just to save a line or 2 - however it will make `holdoutKey` to be `undefined` and perhaps you explicitly want it to be `null`

Reply:
Ah yeah, I'm thinking I want it to be `null`, since that is also how empty values are represented in JSON

I think I could do `matchingHoldout?.holdoutKey ?? null;`